web/packages/teleterm/src/mainProcess/windowsManager.ts

```diff
@@ -83,11 +105,27 @@ export class WindowsManager {
 
     window.once('close', () => {
       this.saveWindowState(window);
+      this.frontendAppInit.reject(
+        new Error('Window was closed before frontend app got initialized')
+      );
     });
 
     // shows the window when the DOM is ready, so we don't have a brief flash of a blank screen
     window.once('ready-to-show', window.show);
 
+    ipcMain.once(
+      'windows-manager-signal-frontend-app-initialization',
```

Comment:
Previously we'd only ever subscribe to events on `ipcMain` [from within `MainProcess`](https://github.com/gravitational/teleport/blob/37e52dc8a8e2c60c710cea3d3c8e78dedbb18d5d/web/packages/teleterm/src/mainProcess/mainProcess.ts#L203).

However, I wanted to use the IPC message to encapsulate the frontend app readiness logic within `WindowsManager`. Since the frontend app won't ever get launched until `WindowsManager.createWindow` gets called, I decided to add the `ipcMain` listener here instead of coming up with some novel way of passing information between `WindowsManager` and `MainProcess`.